### PR TITLE
[RunDB] Move SQLDB to API

### PR DESCRIPTION
**Pull request by alonmr**
On server side we use `SQLDB` as a proxy to the real sqldb so that we won't call ourselves when taking actions against the db like the httpdb does (calls the api from the client).
Moved the `SQLDB` (Now `SQLRunDB`) to the API and used [dependency-injector](https://python-dependency-injector.ets-labs.org/index.html) package to override the client side rundb.
In the future we would like to remove the `SQLRunDB` and use `SQLDB` as the API's RunDB but it requires interface changes.

### CODE REVIEW

File: mlrun/db/factory.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        self._last_db_url = url
+
+        if "://" not in str(url):
+            logger.warning(
+                "Could not detect path to API server, not connected to API server!"
+            )
+            logger.warning(
+                "MLRUN_DBPATH is not set. Set this environment variable to the URL of the API server"
+                " in order to connect"
+            )
```
Comment (TomerShor) [inline mlrun/db/factory.py:53]: Why is this logged even though this env var is not checked here?

File: mlrun/api/rundb/sqldb.py [hunk 2]
```diff
@@ -909,3 +836,9 @@ def verify_authorization(
         # on server side authorization is done in endpoint anyway, so for server side we can "pass" on check
         # done from ingest()
         pass
+
+
+# Once this file is imported it will override the default RunDB implementation (RunDBContainer)
+@containers.override(mlrun.db.factory.RunDBContainer)
+class SQLRunDBContainer(containers.DeclarativeContainer):
+    run_db = providers.Factory(SQLRunDB)
```
Comment (TomerShor) [inline mlrun/api/rundb/sqldb.py:844]: I would move this to the top of the file, as it is related to importing it.
Just to make it a bit clearer

**Review by TomerShor — COMMENTED** [review]
Very nice and a much better organized code 👍 
Minor comments

File: mlrun/db/factory.py (hunk 1 above)
Comment (alonmr) [reply mlrun/db/factory.py:53]: it is checked in 
`url = mlrun.db.get_or_set_dburl("./")`
line 34

File: mlrun/api/rundb/sqldb.py (hunk 2 above)
Comment (alonmr) [reply mlrun/api/rundb/sqldb.py:844]: Can't because need to define SQLRunDB 1st

File: mlrun/db/factory.py (hunk 1 above)
Comment (TomerShor) [reply mlrun/db/factory.py]: but this is only called if `not url`.
if `url != "" and "://" not in str(url)` - the log is not necessarily true.

**Review by TomerShor — APPROVED** [review]

**Pull request merged**